.changeset/weak-trees-perform.md

```diff
@@ -0,0 +1,5 @@
+---
+'mermaid': patch
+---
+
+In State Diagram refactored getDirection to retrieve the direction from rootDoc, defaulting to DEFAULT_DIAGRAM_DIRECTION and Updated setDirection to modify the existing direction if found or prepend a new entry if not.
```

Comment:
Changeset can be brief, if someone wants to get details, the PR will be linked in changelog. 

```suggestion
fix: `getDirection` and `setDirection` in `stateDb` refactored to return and set actual direction
```